main.go

```diff
@@ -351,3 +307,88 @@ func uploadFile(uploadURL string, archiveFilePath string) error {
 
 	return nil
 }
+
+func processStepResult(step *toolresults.Step) string {
+	outcome := step.Outcome.Summary
+
+	switch outcome {
+	case "success":
+		outcome = colorstring.Green(outcome)
+	case "failure":
+		if step.Outcome.FailureDetail != nil {
+			if step.Outcome.FailureDetail.Crashed {
+				outcome += "(Crashed)"
+			}
+			if step.Outcome.FailureDetail.NotInstalled {
+				outcome += "(NotInstalled)"
+			}
+			if step.Outcome.FailureDetail.OtherNativeCrash {
+				outcome += "(OtherNativeCrash)"
+			}
+			if step.Outcome.FailureDetail.TimedOut {
+				outcome += "(TimedOut)"
+			}
+			if step.Outcome.FailureDetail.UnableToCrawl {
+				outcome += "(UnableToCrawl)"
+			}
+		}
+		outcome = colorstring.Red(outcome)
+	case "inconclusive":
+		if step.Outcome.InconclusiveDetail != nil {
+			if step.Outcome.InconclusiveDetail.AbortedByUser {
+				outcome += "(AbortedByUser)"
+			}
+			if step.Outcome.InconclusiveDetail.InfrastructureFailure {
+				outcome += "(InfrastructureFailure)"
+			}
+		}
+		outcome = colorstring.Yellow(outcome)
+	case "skipped":
+		if step.Outcome.SkippedDetail != nil {
+			if step.Outcome.SkippedDetail.IncompatibleAppVersion {
+				outcome += "(IncompatibleAppVersion)"
+			}
+			if step.Outcome.SkippedDetail.IncompatibleArchitecture {
+				outcome += "(IncompatibleArchitecture)"
+			}
+			if step.Outcome.SkippedDetail.IncompatibleDevice {
+				outcome += "(IncompatibleDevice)"
+			}
+		}
+		outcome = colorstring.Blue(outcome)
+	}
+	return outcome
+}
+
+func GetSuccessOfExecution(steps []*toolresults.Step) (bool, error) {
```

Comment:
Does this need to be a public function? It is used in the same package so I think we can make this private.

Reply:
I made it public for the unit tests to be able to use the function. To keep it public and make sense, I pulled it into a step package